src/ApiPort.VisualStudio/Analyze/ProjectAnalyzer.cs

```diff
@@ -79,21 +105,43 @@ private IEnumerable<string> GetAllAssembliesInDirectory(string directory)
             }
         }
 
-        private void DisplaySourceItemsInErrorList(IEnumerable<ISourceMappedItem> items, ICollection<Project> projects)
+        private async Task DisplaySourceItemsInErrorList(IEnumerable<ISourceMappedItem> items, ICollection<Project> projects)
         {
             if (!items.Any())
             {
                 return;
             }
 
+            await VisualStudio.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
             _errorList.Tasks.Clear();
             _errorList.Refresh();
             _errorList.SuspendRefresh();
 
+            var projectWithOutputMappings = new ConcurrentDictionary<string, IVsHierarchy>();
+
+            foreach (var project in projects)
+            {
+                var outputs = await project.GetBuildOutputFilesAsync();
+
+                if (outputs == null)
+                {
+                    continue;
+                }
+
+                var hierarchy = project.GetHierarchy();
+
+                foreach (var output in outputs)
+                {
+                    projectWithOutputMappings.AddOrUpdate(output, hierarchy, (existingKey, existingValue) => hierarchy);
+                }
+            }
+
+            await VisualStudio.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Do we need this again? I would expect us to still be on the main thread due to 115. The only await we have here doesn't have `ConfigureAwait` so it should be on the same context